File: examples/athletic-analyst-ts/src/test-config.ts
/**
 * Test infrastructure — centralizes test-mode flags and helpers.
 * Active only when ATHLETIC_ANALYST_TEST_MODE=1 (used by the integration test runner).
 */

import type { ExecutionSummary } from "./mcp-bridge.js";

export const TEST_MODE = process.env.ATHLETIC_ANALYST_TEST_MODE === "1";

export const EXECUTION_SUMMARY_PREFIX = "EXECUTION_SUMMARY: ";

export function emitTestSummary(summary: ExecutionSummary): void {
  if (!TEST_MODE) return;
  console.error(`${EXECUTION_SUMMARY_PREFIX}${JSON.stringify(summary)}`);
}

File: examples/athletic-analyst-py/src/test_config.py
"""Test infrastructure — centralizes test-mode flags and helpers.

Active only when ATHLETIC_ANALYST_TEST_MODE=1 (used by the integration test runner).
"""

import json
import os
import sys

TEST_MODE = os.getenv("ATHLETIC_ANALYST_TEST_MODE") == "1"

EXECUTION_SUMMARY_PREFIX = "EXECUTION_SUMMARY: "


def emit_test_summary(summary: dict) -> None:
    if not TEST_MODE:
        return
    print(
        f"{EXECUTION_SUMMARY_PREFIX}{json.dumps(summary)}",
        file=sys.stderr,
        flush=True,
    )
